apps/services/endorsements/api/scripts/updateMetadata.ts

```diff
@@ -36,6 +53,89 @@ const limiter = new Bottleneck({
   maxConcurrent: Number(process.env.MAX_CONCURRENT) || 10,
 })
 
+// AUTH
+const acquireAuthToken = async (): Promise<string> => {
+  const tokenOptions = {
+    client_id: environment.endorsementClient.clientId,
+    grant_type: 'client_credentials',
+    scope: GenericScope.system,
+    client_secret: environment.endorsementClient.clientSecret,
+  }
+  const response = await fetch(`${environment.auth.issuer}/connect/token`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: new URLSearchParams(tokenOptions),
+  })
+
+  if (!response.ok) {
+    throw new Error('Failed to acquire access token')
+  }
+
+  const result = await response.json()
+
+  return result.access_token
+}
+
+// taken from @island.is/auth-nest-tools
+const parseScopes = (scopes: undefined | string | string[]): string[] => {
+  if (scopes === undefined) {
+    return []
+  }
+  if (typeof scopes === 'string') {
+    return scopes.split(' ')
+  }
+  return scopes
+}
+
+// token is not validated at this point, we expect the underlying services to authenticate the token
+const convertTokenToAuth = (token: string): Auth => {
+  try {
+    const payload = token.split('.')[1]
+    const decodedPayload: JwtPayload = JSON.parse(
+      Buffer.from(payload, 'base64').toString(),
+    )
+
+    return {
+      scope: parseScopes(decodedPayload.scope),
+      client: decodedPayload.client_id,
+      authorization: token,
+    }
+  } catch (e) {
+    logger.error('Failed to convert token to authentication')
+    throw e
+  }
+}
```

Comment:
Although I love the bruteforceness of parsing the token this way we'll upgrade to using `jsonwebtoken's decode` 😄 

We ran into some issues with token exchange since client credential flow can't request openid scope.
The auth process seems to work in some cases but is not reliable enough to allow the script to complete all it's tasks.
This script is not crucial for our current objectives in the endorsement system and has been disabled until we can allocate more time to flush out the script auth flow.

We'll definitely add the tooling to `auth-nest-tools` when we feel confident in the flow 😃